pkg/util/hardware/mem_info.go

```diff
@@ -0,0 +1,43 @@
+//go:build !darwin && !fressbsd && !openbsd
+// +build !darwin,!fressbsd,!openbsd
+
+package hardware
+
+import (
+	"os"
+
+	"github.com/milvus-io/milvus/pkg/log"
+	"github.com/shirou/gopsutil/process"
+	"go.uber.org/zap"
+)
+
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// GetUsedMemoryCount returns the memory usage in bytes.
+func GetUsedMemoryCount() uint64 {
+	proc, err := process.NewProcess(int32(os.Getpid()))
```

Comment:
how long does it take to run this command?

Do we need to cache the memory usage?

Reply:
it's fast (about 100μs, including reading process and memory info)

Reply:
cache the process itself is a good idea, the memory usage has to be parsed every time (it won't be updated after reading)